packages/cli/src/UserManagement/email/NodeMailer.ts

```diff
@@ -1,87 +1,52 @@
-/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+import { Service } from 'typedi';
+import { pick } from 'lodash';
 import type { Transporter } from 'nodemailer';
 import { createTransport } from 'nodemailer';
 import type SMTPConnection from 'nodemailer/lib/smtp-connection';
-import { Service } from 'typedi';
+import { GlobalConfig } from '@n8n/config';
 import { ErrorReporterProxy as ErrorReporter } from 'n8n-workflow';
-import config from '@/config';
-import type { MailData, SendEmailResult } from './Interfaces';
+
 import { Logger } from '@/Logger';
+import type { MailData, SendEmailResult } from './Interfaces';
 
 @Service()
 export class NodeMailer {
-	private transport?: Transporter;
+	readonly sender: string;
 
-	constructor(private readonly logger: Logger) {}
+	private transport: Transporter;
 
-	async init(): Promise<void> {
-		const transportConfig: SMTPConnection.Options = {
-			host: config.getEnv('userManagement.emails.smtp.host'),
-			port: config.getEnv('userManagement.emails.smtp.port'),
-			secure: config.getEnv('userManagement.emails.smtp.secure'),
-			ignoreTLS: !config.getEnv('userManagement.emails.smtp.startTLS'),
-		};
+	constructor(
+		globalConfig: GlobalConfig,
+		private readonly logger: Logger,
+	) {
+		const smtpConfig = globalConfig.userManagement.emails.smtp;
```

Comment:
Can a consumer of `globalConfig` mutate it by accident? If so should we prevent this?

Reply:
right now, unfortunately yes. The plan is to switch to read-only proxies for the config eventually so that we can 
1. prevent writes
2. block any callers from outside our packages from reading any config

Created ENG-134 for that.